yt_dlp/extractor/elemental_tv.py

```diff
@@ -0,0 +1,107 @@
+import json
+import re
+import time
+
+from ..extractor.common import InfoExtractor
+from ..networking.exceptions import HTTPError
+from ..utils import ExtractorError
+
+
+class ElementalTVIE(InfoExtractor):
+    _LOGIN_REQUIRED = True
+    _NETRC_MACHINE = 'elemental_tv'
+    _VALID_URL = r'https?://play\.elemental\.tv/channel/[0-9a-f]{24}'
+    _TESTS = [{
+        'url': 'https://play.elemental.tv/channel/573f5a14761973ec1d502507',
+        'info_dict': {
+            'id': '573f5a14761973ec1d502507',
+            'ext': 'mp4',
+            'title': 'БНТ 1 HD',
+            'thumbnail': 'https://play.elemental.tv/v1/tumblrs/573f5a14761973ec1d502507',
+            'age_limit': 0,
+        },
+    }]
+
+    access_token = ''
+    channel_id = ''
+
+    def _get_channel_id(self, url):
+        url_parts = re.search('(?<=channel/)[0-9a-f]{24}', url)
+
+        if not url_parts or not url_parts.group(0):
+            return None
+
+        self.channel_id = url_parts.group(0)
+        self.write_debug(f'Channel ID: {self.channel_id}')
+
+        if not self.channel_id:
+            raise ExtractorError('Channel ID not found')
+
+    def _get_stream_metadata(self):
+        try:
+            headers = {
+                'Authorization': 'Bearer ' + self.access_token,
+            }
+            res_api = self._download_json(
+                'https://play.elemental.tv/v1/channels', self.channel_id, headers=headers)
+            data = res_api.get('data').get(self.channel_id)
+
+            if not data:
+                self.write_debug('Getting metadata failed')
+                return {}
+
+            return {
+                'title': data.get('name'),
+                'age_limit': data.get('age'),
+                'thumbnail': data.get('tumblrurl'),
+            }
+        except Exception:
+            self.write_debug('Getting metadata failed')
+            return {}
+
+    def _get_stream_url(self):
+        # Stream URL needs current epoch time rounded to 10000s
+        begin = int((time.time() - 60) / 10000) * 10000
```

Comment:
Is this based on some of the site's javascript? `current_time` should look back two hours? A reference url would be nice to ensure correctness and for future maintenance